code/datums/components/cult_status.dm

```diff
@@ -0,0 +1,127 @@
+/**
+ * # Cult Status component
+ *
+ * Component used to track and display the visual cult effects.
+ *
+ * Used to apply and remove the traits, and overlays associated with the growing size of a cult.
+ */
+/datum/component/cult_status
+	///Which stage of visibilty is the cultist at?
+	var/stage = STAGE_CULT_UNSEEN
+	///Have the eyes started glowing?
+	var/glowing_eyes = FALSE
+	///Has the halo been added?
+	var/has_halo = FALSE
+
+/**
+ * Check if the cultist needs to have their
+ */
+/datum/component/cult_status/Initialize()
+	RegisterSignal(parent, COMSIG_CULT_VIS, .proc/raise_level)
```

Comment:
Procnals are generally discouraged. Could you make this a global signal when the status of the cult changes and have the halos decide what stage to take?